ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobs.tsx

```diff
@@ -0,0 +1,220 @@
+    const { reportSnapshots, isLoading, error } = useFetchReportHistory({
+        id: reportId,
+        query: '',
+        page,
+        perPage,
+    });
+
+    const handleChange = (checked: boolean) => {
+        setShowOnlyMyJobs(checked);
+    };
+
+    if (isLoading) {
+        return (
+            <Bullseye>
+                <Spinner isSVG />
+            </Bullseye>
+        );
+    }
+
+    if (error) {
+        return (
+            <NotFoundMessage
+                title="Error fetching report history"
+                message={error || 'No data available'}
+            />
+        );
+    }
+
+    if (!reportSnapshots.length) {
+        return (
+            <Bullseye>
+                <EmptyStateTemplate title="No run history" headingLevel="h2" icon={CubesIcon} />
+            </Bullseye>
+        );
+    }
+
+    return (
+        <>
+            <Toolbar>
+                <ToolbarContent>
+                    <ToolbarItem>
+                        <CheckboxSelect
+                            ariaLabel="CVE severity checkbox select"
+                            toggleIcon={<FilterIcon />}
+                            selections={filteredStatuses}
+                            onChange={setFilteredStatuses}
+                            placeholderText="Filter by status"
+                        >
+                            <SelectOption value="PREPARING">Preparing</SelectOption>
+                            <SelectOption value="WAITING">Waiting</SelectOption>
+                            <SelectOption value="SUCCESS">Successful</SelectOption>
+                            <SelectOption value="ERROR">Error</SelectOption>
+                        </CheckboxSelect>
+                    </ToolbarItem>
+                    <ToolbarItem className="pf-u-flex-grow-1">
+                        <Switch
+                            id="view-only-my-jobs"
+                            label="View only my jobs"
+                            labelOff="View only my jobs"
+                            isChecked={showOnlyMyJobs}
+                            onChange={handleChange}
+                        />
+                    </ToolbarItem>
+                    <ToolbarItem variant="pagination" alignment={{ default: 'alignRight' }}>
+                        <Pagination
+                            toggleTemplate={({ firstIndex, lastIndex }) => (
+                                <span>
+                                    <b>
+                                        {firstIndex} - {lastIndex}
+                                    </b>{' '}
+                                    of <b>many</b>
+                                </span>
+                            )}
+                            page={page}
+                            perPage={perPage}
+                            onSetPage={(_, newPage) => setPage(newPage)}
+                            onPerPageSelect={(_, newPerPage) => setPerPage(newPerPage)}
+                            isCompact
```

Comment:
`isCompact` is definitely the way to go here when using the "of many" pagination pattern, IMO. The Listening Endpoints page currently uses the full Pagination component, but the additional controls aren't useful in that context.